src/realm/query_expression.hpp

```diff
@@ -4531,13 +4623,23 @@ class Compare : public CompareBase {
         ValueBase* right = m_right_const_values ? m_right_const_values : &right_buf;
 
         for (; start < end;) {
-            if (!m_left_const_values)
-                m_left->evaluate(start, left_buf);
-            if (!m_right_const_values)
-                m_right->evaluate(start, right_buf);
-            match = ValueBase::template compare<TCond>(*left, *right, left_cmp_type, right_cmp_type);
-            if (match != not_found && match + start < end)
-                return start + match;
+            // In case of wildcard query strings, we will get a value for every collection matching the path
+            // We need to match those separately against the other value - which might also come in multiple
+            // instances.
+            Subexpr::Index right_index(start);
```

Comment:
This looks optimal to me, but since it is such a hot path, could you check how much performance is lost on normal queries that don't have any `more()`? I think the cost is two more conditionals per evaluation, so hopefully not too bad.

Reply:
If you have a query comparing 2 integer properties, you will have a 2% degradation of performance. On the other hand, I think this change will speed up queries over linklists as we will only fetch values until we have a match. And we are also avoiding some copying to intermediate buffers.